### ISSUE TITLE
Parsing floats in URDF yields wrong value depending on locale

### ISSUE THREAD

**Issue by dbasa** [ros/urdfdom_headers#41]
`urdfdom_headers` uses `std::stod` to parse floating point values like position (xyz) and rotation (rpy) from URDF files. The result of `std::stod` is locale dependant. So with e.g. a German locale (de_DE.UTF-8) "1.57079632679" is parsed to `1`.
This is the case in  `urdfdom_headers/urdf_model/include/urdf_model/pose.h` Line 70 (not necessarily exclusive)

In my opinion the issue described [here](https://github.com/ros-visualization/rviz/issues/1151) can be linked to this error, as can be the issue described [here](https://github.com/ros-planning/moveit/issues/690).

[This link](https://www.reddit.com/r/cpp/comments/2e68nd/stdstod_is_locale_dependant_but_the_docs_does_not/) states that "[...] using a stringstream to convert it worked perfectly fine."

If desired I can fix this issue using stringstream and create a pull request, but I have to admit I have no background knowledge in ros package building and testing procedures.

Cheers

**Comment by traversaro** [comment]
Related urdfdom issue: https://github.com/ros/urdfdom/issues/98 .

**Comment by clalancette** [comment]
So I ended up looking into this a bit.  @traversaro did some good investigating of the issue in https://github.com/robotology/idyntree/issues/288#issuecomment-299447873, and digging down into the URDF schema, this is represented as an xs:double.  Looking at the XSD document [here](https://www.w3.org/TR/xmlschema-2/#double), it says that the mantissa for a double is represented as a "decimal" number.  And looking at the XSD document [here](https://www.w3.org/TR/xmlschema-2/#decimal), a decimal number is a "finite-length sequence of decimal digits (#x30-#x39) separated by a period as a decimal indicator".  Thus, the locale should have no effect on how the document is parsed, and using `str::stod` is incorrect.  I've prepared a patch for urdfdom_headers which switches to using the stream operator; I'll open that presently.
